crates/pink/src/storage.rs

```diff
@@ -57,21 +73,20 @@ where
             let r = crate::runtime::using_mode(mode, f);
             (r, crate::runtime::get_side_effects())
         });
-        if rollback {
-            self.overlay.rollback_transaction()
-        } else {
-            self.overlay.commit_transaction()
+        overlay
+            .commit_transaction()
+            .expect("BUG: mis-paired transaction");
+        if !rollback {
```

Comment:
Do we need to call `commit_transaction()` if it's rolled back

Reply:
I think either commit or rallback should be called to assert the nested transactions are correctly paired.